designdocs/batched_messages.md

```diff
@@ -10,98 +10,235 @@
+user-created messages.
+
+One risk is that providers on MRC could simply register a new schema and
+announce batches "unofficially". We have not decided whether or not to let everyone
+with enough token balance register a schema. Other MRC participants would need to
+first learn about and evaluate new schemas, then update their software to
+consume a new message type.
+
+There are some upsides to deriving batching logically from existing structures.
+One is cost savings. Not having a batch structure means we don't need to worry
+about any on-chain computation associated with batch messages -- we simply look
+at the format and location on the parent schema and we can deduce whether the
+file is singular or plural.
 
 ### Alternatives and Rationale
-We discussed whether a batch message itself can be delegated, but this would have complicated things and we cannot come up with a use case for delegating batches. It also violates the idea of users delegating explicitly to every provider that performs a service for them, which is a fundamental value we want to apply to the network.
+We discussed whether a batch message itself can be delegated, but this would
+have complicated things and we cannot come up with a use case for delegating
+batches. It also violates the idea of users delegating explicitly to every
+provider that performs a service for them, which is a fundamental value we want
+to apply to the network.
 
 We discussed whether to allow URLs such as HTTP/HTTPS or other URLs and instead opted for content-addressable URIs (CIDv1) which can be resolved by some other service.  This allows us to put the file hash directly into a URI.  It reduces message storage because we don't have to include both a URL and a file hash. A file hash is necessary as a check against file tampering.
 
 We revisited the idea of whether it really is necessary to include a file size. We will be charging a premium for larger files, however, there will be per-byte discount for larger files in order to create an incentive for posting batches while reducing the incentive for announcers to allow spam. Although the processing and downloading time for enormous files also serves as a disincentive for spam, we feel it would not be sufficient.
 
-Despite the fact that announcers can lie abut the file size, the file_size parameter also serves as an on-chain declaration that not only allows consumers of batches to quickly discover if a batch announcer was honest, but the file requestor can know in advance when to stop requesting data.
+Despite the fact that announcers can lie about the file size, the file_size
+parameter also serves as an on-chain declaration that not only allows consumers
+of batches to quickly discover if a batch announcer was honest, but the file
+requestor can know in advance when to stop requesting data.
+
+#### Message Payloads
+The document above describes payload location as an enumeration of on-chain and
```

Comment:
I wonder if since these implementation details are so up in the air whether it's worth outlining them at all. Maybe we could just say that the batch/schema updates will affect the current message retrieval format and that the specifics are TBD by implementer(s).  It's a good suggestion but maybe belongs in a story for that instead (i.e. don't just delete the ideas)